### ISSUE TITLE
`exporters/otlp/otlptrace/otlptrace{grpc,http}` defaults to localhost insecure port but without insecure option

### ISSUE THREAD

**Issue by Jorropo** [open-telemetry/opentelemetry-go#4803]
### Environment

- OS: `linux`
- Architecture: `amd64/v3`
- Go Version: `go1.22-988b718f41`
- opentelemetry-go version: `v1.21.0`

### Steps To Reproduce

I am using copy pasted from [here](https://www.jaegertracing.io/docs/1.52/deployment/#all-in-one):
```bash
docker run -d --rm --name jaeger \
  -e COLLECTOR_OTLP_ENABLED=true \
  -e COLLECTOR_ZIPKIN_HOST_PORT=:9411 \
  -p 5775:5775/udp \
  -p 6831:6831/udp \
  -p 6832:6832/udp \
  -p 5778:5778 \
  -p 16686:16686 \
  -p 14250:14250 \
  -p 14268:14268 \
  -p 14269:14269 \
  -p 4317:4317 \
  -p 4318:4318 \
  -p 9411:9411 \
  jaegertracing/all-in-one:1.52
```
Then on this branch: https://github.com/Jorropo/go-ipfs/tree/otlp-repro
```bash
go install ./cmd/ipfs
OTEL_TRACES_EXPORTER=otlp ipfs daemon --init
```
or
```bash
OTEL_EXPORTER_OTLP_PROTOCOL=grpc OTEL_TRACES_EXPORTER=otlp ipfs daemon --init
```

---

https://github.com/open-telemetry/opentelemetry-go/blob/27f70a335008ed7cdb398013eedf6812cc1c44e0/exporters/otlp/otlptrace/otlptracehttp/internal/otlpconfig/options.go#L114

Defaults to `localhost:4318`.
So does grpc one

https://github.com/open-telemetry/opentelemetry-go/blob/27f70a335008ed7cdb398013eedf6812cc1c44e0/exporters/otlp/otlptrace/otlptracegrpc/internal/otlpconfig/options.go#L114

However insecure defaults to `false`.

---

So trying to use this I get (http):
```
2023/12/29 08:11:50 traces export: Post "https://localhost:4318/v1/traces": http: server gave HTTP response to HTTPS client
```
(grpc):
```
2023/12/29 08:17:50 traces export: context deadline exceeded: rpc error: code = Unavailable desc = connection error: desc = "transport: authentication handshake failed: tls: first record does not look like a TLS handshake"
```

### Expected behavior

Defaults should be `http://localhost:4318`, I think it is good to default to `https://` for non `localhost`, `127.0.0.1` and `::1` hosts but I don't care personally.

I guess it could also be an issue with jaeger docs that incorrectly handle the insecure port, I don't know.

**Comment by pellared** [comment]
Duplicates https://github.com/open-telemetry/opentelemetry-go/issues/4147

TL;DR; This would be a breaking change.

**Comment by Jorropo** [comment]
@pellared Ok thx, I didn't found #4147.

Could we hint the user at `OTEL_EXPORTER_OTLP_INSECURE=true` if they are using localhost and getting tls error then ?

**Comment by pellared** [comment]
> Could we hint the user at OTEL_EXPORTER_OTLP_INSECURE=true in the log message if they are using localhost and getting tls error then ?

Sure. Are you able to create a PR which does it?
